server/cloudapps/proxy.go

```diff
@@ -0,0 +1,76 @@
+// Copyright (c) 2020-present Mattermost, Inc. All Rights Reserved.
+// See License for license information.
+
+package cloudapps
+
+import (
+	"encoding/json"
+	"io"
+	"net/http"
+	"path"
+
+	pluginapi "github.com/mattermost/mattermost-plugin-api"
+	"github.com/mattermost/mattermost-server/v5/model"
+	"github.com/mattermost/mattermost-server/v5/plugin"
+
+	"github.com/mattermost/mattermost-plugin-cloudapps/server/configurator"
+)
+
+type Proxy interface {
+	OnUserJoinedChannel(pluginContext *plugin.Context, channelMember *model.ChannelMember, actor *model.User)
+}
+
+type proxy struct {
+	configurator.Configurator
+	mm       *pluginapi.Client
+	registry Registry
+	Subscriptions
+}
+
+var _ Proxy = (*proxy)(nil)
+
+func NewProxy(mm *pluginapi.Client, configurator configurator.Configurator, subs Subscriptions) Proxy {
+	return &proxy{
+		Configurator:  configurator,
+		Subscriptions: subs,
+		mm:            mm,
+	}
+}
+
+func (p *proxy) SendChangeNotification(s *Subscription, msg interface{}) {
+	client, err := p.GetNotificationClient(s.AppID)
+	if err != nil {
+		// <><> TODO log
+		return
+	}
+
+	app, err := p.registry.GetApp(s.AppID)
+	if err != nil {
+		// <><> TODO log
+		return
+	}
+
+	piper, pipew := io.Pipe()
+	go func() {
+		defer pipew.Close()
+		encodeErr := json.NewEncoder(pipew).Encode(msg)
+		if encodeErr != nil {
+			// <><> TODO log
+			return
+		}
+	}()
+
+	u := path.Join(app.RootURL, "notify", string(SubjectUserJoinedChannel))
```

Comment:
Should we force the path or leave the app to handle its own paths? e.g. An app may want to handle the wishes at `root/wish/certainWish`but handle all the notifications at `root/notifications`.

Reply:
Right. Ack the dilemma. For the POC I'm 1/5 on hardcoding forced paths to try to work out a convention.